src/mixins/search-mixin.ts

```diff
@@ -115,9 +114,11 @@ export class SearchMixin extends Mixins(CommonMixin) {
   set entity_type_cd (type: EntityTypes) {
     // special case for sub-menu
     if (type === EntityTypes.INFO) {
-      // set empty value until user chooses a new one
+      // set empty values until user chooses a new one
       // (don't use null in case it's already null as we want reactivity)
       this.setEntityTypeCd('')
+      this.setSearchCompanyType('')
+      // this.setLocation(null) // *** TODO: fix bug (Severin)
```

Comment:
This is a bug partially fixed.

Basically, because of the reverse flow, if you select an xpro aml and then you select a jurisdiction and then you go back to the entity type menu, some things need to be cleared.

I clear these for all flows because the other flows don't care.